bokehjs/src/lib/core/has_props.ts

```diff
@@ -536,8 +522,12 @@ export abstract class HasProps extends Signalable() implements Equatable, Printa
 
   attach_document(doc: Document): void {
     // This should only be called by the Document implementation to set the document field
-    if (this.document != null && this.document != doc)
-      throw new Error("models must be owned by only a single document")
+    if (this.document != null) {
+      if (this.document == doc)
+        return
+      else
+        throw new Error("models must be owned by only a single document")
+    }
```

Comment:
Why prefer a nested conditional here?

Reply:
I'm not sure how else would you write this.